Controller/Adminhtml/Menu/Save.php

```diff
@@ -102,107 +102,102 @@ public function execute()
         $menu->saveStores($this->getRequest()->getParam('stores'));
         $nodes = $this->getRequest()->getParam('serialized_nodes');
 
-        if (!empty($nodes)) {
-            $nodes = json_decode($nodes, true);
-            $nodes = $this->_convertTree($nodes, '#');
+        $filterBuilder = $this->filterBuilderFactory->create();
```

Comment:
Could you please extract all this get node collection logic into a separated private method?